### PR TITLE
Feature/file selection

### PR DESCRIPTION
**Pull request by pvrobays**
added the new file type, which is able to multi-select files. See the extended example preferences for config.

### CODE REVIEW

**Comment by pvrobays** [comment]
TODO:
- [ ] Test
- [ ] add usage to README
- [ ] update CHANGELOG

**Comment by pvrobays** [comment]
closes #55

File: src/app/components/main/components/group/components/fields/file/index.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+            if (!result)
+                return;
+
+            if (result.length) {
+                this.onChange(multiSelections ? result : result[0]);
+            }
+        }
+
+        const { multiSelections, value, help, label } = this;
```
Comment (lacymorrow) [inline src/app/components/main/components/group/components/fields/file/index.js:39]: You have two variables named `multiSelections`, they're in two different scopes but one scope receives both variables, plus it's confusing in this context.

You can just define the vars at the top of the render method and use them throughout.

**Review by lacymorrow — CHANGES_REQUESTED** [review]
Looks great! There are a few changes I requested but overall an awesome addition

File: src/app/utils/isArray.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+﻿export function isArray(obj) {
+    return Array.isArray(obj);
```
[the remark this answers is not in the document]
Comment (pvrobays) [reply src/app/utils/isArray.js:2]: I know, my most helpful util function ever! 😅

**Comment by pvrobays** [comment]
@lacymorrow FYI. Fixed your PR remarks and added the ability to have multi selection for a directory as well.
Closes #81

**Review by lacymorrow — APPROVED** [review]
LGTM++

**Comment by pvrobays** [comment]
Merged. FYI @lacymorrow I've had some issues with my OCD with the linting settings, changed these things:
- No spaces between parentheses
- Always end with semicolon

**Pull request merged**